src/tests/ie_test_utils/common_test_utils/data_utils.hpp

```diff
@@ -195,16 +195,22 @@ template<class T>
 void inline
 fill_data_random(T *pointer, std::size_t size, const uint32_t range = 10, int32_t start_from = 0, const int32_t k = 1,
                  const int seed = 1) {
+    if (range == 0) {
+        for (std::size_t i = 0; i < size; i++) {
+            pointer[i] = static_cast<T>(start_from);
+        }
+        return;
+    }
+
     testing::internal::Random random(seed);
     const uint32_t k_range = k * range; // range with respect to k
     random.Generate(k_range);
 
     if (start_from < 0 && !std::is_signed<T>::value) {
         start_from = 0;
     }
-
     for (std::size_t i = 0; i < size; i++) {
-        pointer[i] = static_cast<T>(start_from + static_cast<T>(random.Generate(k_range)) / k);
+        pointer[i] = static_cast<T>((start_from + static_cast<T>(random.Generate(k_range))) / k);
```

Comment:
Now this function is not trivial, and requires some preprocessing of the start_from argument (I would propose to add a description comment).
Is it possible to allow to specify start_from as a float value instead? Or this would require much more changes?

Reply:
from high level nothing does change (check the code). Generators work with int numbers. My suggestion is use such construction. Moreover, the changes which we merged before is not so trivial :)